pyClient/zeth/joinsplit.py

```diff
@@ -601,46 +513,55 @@ def _compute_proof_hashes(proof_json: GenericProof) -> Tuple[bytes, bytes]:
         if key != "inputs":
             proof_elements.extend(proof_json[key])
     return (
-        sha256(encode_message_to_bytes(proof_elements)).digest(),
-        sha256(encode_pub_input_to_hash(proof_json["inputs"])).digest())
+        encode_message_to_bytes(proof_elements),
+        encode_message_to_bytes(proof_json["inputs"]))
 
 
 def sign_mix_tx(
+        signing_keypair: SigningKeyPair,  # Ephemeral signing key, tied to proof
         sender_eph_pk: EncryptionPublicKey,  # Ephemeral key used for encryption
         ciphertexts: List[bytes],  # Encyrpted output notes
         proof_json: GenericProof,  # Proof for the mix transaction
-        signing_keypair: SigningKeyPair  # Ephemeral signing key, tied to proof
 ) -> int:
+    """
+    Generate a Schnorr signature on the hash of the ciphertexts, proofs and
+    primary inputs. This is used to solve transaction malleability.  We chose
+    to sign the hash and not the values themselves for modularity (to use the
+    same code regardless of whether GROTH16 or PGHR13 proof system is chosen),
+    and sign the hash of the ciphers and inputs for consistency.
+    """
     assert len(ciphertexts) == constants.JS_INPUTS
-    # Hash the pk_sender and cipher-texts
-    sender_eph_pk_bytes = encode_encryption_public_key(sender_eph_pk)
-    ciphers = sender_eph_pk_bytes + ciphertexts[0] + ciphertexts[1]
-    hash_ciphers = sha256(ciphers).digest()
 
-    # Hash the proof
-    proof_hash, pub_inputs_hash = _compute_proof_hashes(proof_json)
+    # The message to sign consists of (in order):
+    #   - senders public encryption key
+    #   - ciphertexts
+    #   - proof elements
+    #   - public input elements
+    h = sha256()
+    h.update(encode_encryption_public_key(sender_eph_pk))
+    for ciphertext in ciphertexts:
+        h.update(ciphertext)
 
-    # Compute the joinSplit signature
-    return _sign(signing_keypair, hash_ciphers, proof_hash, pub_inputs_hash)
+    proof_hash, pub_inputs_hash = _compute_proof_hashes(proof_json)
```

Comment:
perhaps rename this function as we only encode now.

Reply:
Renamed